osi_environment.proto

```diff
@@ -324,4 +336,148 @@ message EnvironmentalConditions
+        //
+        // For cloud determination, the observer will choose a location from which he can overlook the entire sky.
+        // When determining the total cloud cover, all clouds or parts of clouds, regardless of height and genus,
+        // that are above his location will be added. Aircraft contrails are also counted as cloud cover if they remain
+        // in the sky for more than 15 minutes. On the other hand, fog is not counted as cloud cover.
+        //
+        // With a little practice, the observer can easily determine the total amount of cloud cover.
+        // Problems occur only with scattered clouds or many single clouds (cumulus).
+        // Their cloud cover is often overestimated. On the other hand,
+        // the coverage of thin ice clouds (cirrus) is often underestimated [3].
+        //
+        // \par References:
+        // [1] ISO 15469:2004(E) / CIE S 011/E:2003 : Spatial distribution of skylight
+        // [2] https://en.wikipedia.org/wiki/Cloud_cover
+        // [3] https://www.dwd.de/DE/service/lexikon/Functions/glossar.html?lv2=100932&lv3=101016
+        //
+        enum CloudStateLevel
+        {
+            // Cloud State is unknown (must not be used in ground truth).
+            //
+            CLOUD_STATE_LEVEL_UNKNOWN = 0;
+
+            // Other (unspecified but known) CloudState.
+            //
+            CLOUD_STATE_LEVEL_OTHER = 1;
+
+            // Cloudless (0/8)
+            //
+            CLOUD_STATE_LEVEL_CLOUDLESS = 2;
+
+            // Sunny (1/8)
+            //
+            CLOUD_STATE_LEVEL_SUNNY = 3;
+
+            // Serene (2/8))
+            //
+            CLOUD_STATE_LEVEL_SERENE = 4;
+
+            // Slightly Cloudy (3/8)
+            //
+            CLOUD_STATE_LEVEL_SLIGHTLY_CLOUDY = 5;
+
+            // Light Cloudy (4/8)
+            //
+            CLOUD_STATE_LEVEL_LIGHT_CLOUDY = 6;
+
+            // Cloudy (5/8)
+            //
+            CLOUD_STATE_LEVEL_CLOUDY = 7;
+
+            // Heavy Cloudy (6/8)
+            //
+            CLOUD_STATE_LEVEL_HEAVY_CLOUDY = 8;
+
+            // Almost Covered (7/8)
+            //
+            CLOUD_STATE_LEVEL_ALMOST_COVERED = 9;
+
+            // Covered (8/8)
+            //
+            CLOUD_STATE_LEVEL_COVERED = 10;
+        }
+    }
+
+    //
+    // \brief Defines wind properties.
+    //
+    // \note Aligns with the wind speed and direction as defined in ASAM OpenSCENARIO.
+    //
+    message Wind
+    {
+        // The target direction of the wind (not the origin direction) in the ground/xy-plane of the \c
+        // world coordinate system. Corresponds to the heading/yaw angle. \c
+        // x-axis-direction is 0 rad.
+        //
+        // Unit: rad
+        //
+        // The preferred angular range is [-pi, pi]
+        //
+        // \note The range in openSCENARIO is Range [0...2 pi[. Mind the conversion.
+        //
+        optional Vector3d direction = 1;
```

Comment:
**Output CCB 24.11.21:**
Day to day life, the wind **origin** direction is usually given, and not the target direction.
suggestion is to change the name from "direction" to "origin_direction" and update the documentation.
If the direction is not consistent with OpenScenario, then please add a note to indicate this as well.